### PR TITLE
Update for redis 7

### PR DESCRIPTION
**Pull request by AviAvni**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -491,6 +502,9 @@ workflows:
       - build:
           <<: *context
           <<: *not-on-integ-branch
+          matrix:
+            parameters:
+              redis_version: ["6.2", "7.0-rc2"]
```
Comment (rafie) [inline .circleci/config.yml:507]: simply `"7"`.

File: .circleci/config.yml [hunk 2]
```diff
@@ -491,6 +502,9 @@ workflows:
       - build:
           <<: *context
           <<: *not-on-integ-branch
+          matrix:
```
Comment (rafie) [inline .circleci/config.yml:505]: You may want to prepend:
```
name: build-with-redis-<<matrix.redis_version>>
```
so you'll know what's running.

Comment (AviAvni) [reply .circleci/config.yml:505]: the redis_version on the already in the name
```
ci/circleci: build-6.2
ci/circleci: build-7.0-rc2
```

File: .circleci/config.yml (hunk 1 above)
Comment (AviAvni) [reply .circleci/config.yml]: done thanks

**Review by redisfab — DISMISSED** [review]

**Review by rafie — APPROVED** [review]

File: .circleci/config.yml [hunk 3]
```diff
@@ -502,16 +533,43 @@ workflows:
...
+              - platform: centos7
+                redis_version: "7"
+              - platform: centos7
+                redis_version: "unstable"
+              - platform: ol8
+                redis_version: "6.0"
+              - platform: ol8
+                redis_version: "7"
+              - platform: ol8
+                redis_version: "unstable"
```
Comment (rafie) [inline .circleci/config.yml:565]: can't we just check the redis-server matrix for a single platform?

Comment (AviAvni) [reply .circleci/config.yml:565]: yes but we want also to run the other platforms

Comment (rafie) [reply .circleci/config.yml:565]: why? logic is the same for all of them.

File: .circleci/config.yml [hunk 4]
```diff
@@ -502,16 +533,43 @@ workflows:
           <<: *on-integ-and-version-tags
       - platform_build:
           <<: *context
-          <<: *on-integ-and-version-tags
+          <<: *on-any-branch # revert on-integ-and-version-tags
```
Comment (rafie) [inline .circleci/config.yml:536]: need to actually revert

Comment (AviAvni) [reply .circleci/config.yml:536]: will revert when I see all tests can pass

**Review by rafie — APPROVED** [review]

**Review by swilly22 — APPROVED** [review]

**Pull request merged**